tezos/context/src/working_tree/storage.rs

```diff
@@ -398,7 +401,8 @@ pub struct PointerToInodeInner {
     hash_id: B48,
     is_commited: bool,
     inode_id: B31,
-    offset: B64,
+    is_offset_available: bool,
```

Comment:
How does this work?

Reply:
This is to make sure that when we call, `PointerToInodeInner::get_offset`, the method return a valid (previously set) offset.
Without this bit, `get_offset` could return an invalid offset and we would have no error.
https://github.com/tezedge/tezedge/pull/889/files#diff-cfe31f13c183c59ed45a323b4b25dffd667af5d4b4d5d9e5af3cbd0a30a965b0R505

Reply:
I see, makes sense. But wouldn't it be better to use some sentinel value (like `MAX(u64)`) and not lose the bit? or what about 0? it isn't a valid offset right? (these are negative right? not absolute positions in the file)

Reply:
Ah yes, you are right, I will use `0` as sentinel value.

> these are negative right? not absolute positions in the file

Note that here, in `PointerToInodeInner`, the offset is the absolute offset, not relative.

Relative offset are used in the file, but when they are deserialized into the working tree (`PointerToInodeInner` and `DirEntryInner`), they are directly converted to absolute offset

Reply:
> Relative offset are used in the file, but when they are deserialized into the working tree (PointerToInodeInner and DirEntryInner), they are directly converted to absolute offset

Ok, related to this, remember that we don't want to load inodes that are not going to be used unnecessarily, in H that becomes a big issue is not handled properly.